crates/runtime-ffi/src/api.rs

```diff
@@ -25,45 +25,30 @@ static DEPLOY_TEMPLATE_RECEIPT_TYPE: Type = Type::Str("deploy-template receipt")
 static SPAWN_APP_RECEIPT_TYPE: Type = Type::Str("spawn-app receipt");
 static EXEC_APP_RECEIPT_TYPE: Type = Type::Str("exec-app receipt");
 
-macro_rules! maybe_gas {
-    ($gas_metering:expr, $gas_limit:expr) => {{
-        use svm_types::Gas;
-
-        if $gas_metering {
-            Gas::with($gas_limit)
-        } else {
-            Gas::new()
-        }
-    }};
-}
-
-macro_rules! addr_to_svm_byte_array {
-    ($ty:expr, $raw_byte_array:expr, $addr:expr) => {{
-        let (ptr, len, cap) = $addr.into_raw_parts();
-
-        to_svm_byte_array!($ty, $raw_byte_array, ptr, len, cap);
-    }};
-}
-
-macro_rules! vec_to_svm_byte_array {
-    ($ty:expr, $raw_byte_array:expr, $vec:expr) => {{
-        let (ptr, len, cap) = $vec.into_raw_parts();
-
-        to_svm_byte_array!($ty, $raw_byte_array, ptr, len, cap);
-    }};
+#[inline(always)]
+fn maybe_gas(gas_metering: bool, gas_limit: u64) -> Gas {
```

Comment:
since we don't use `gas_metering` for now (we initially thought we would),
maybe a better suitable name would be: `gas_enabled` ?

The `Gas::new()` was meant to ease writing tests that don't care (or want to mess) with anything related to the `Gas`